admin/admin.py

```diff
@@ -89,14 +93,30 @@ def remove_banner() -> Response:
     :returns: Redirect to admin page with status message
     """
     settings = {
-        'banner_enabled': False,
-        'banner_importance': False,
-        'banner_message': ''
+        'banner': {
+            'banner_enabled': False,
+        }
     }
     flash_msg = 'Banner removed successfully'
     return save_settings(settings, flash_msg)
 
 
+@admin_bp.route('/set_theme', methods=['POST'])
+@admin_required
+def set_theme() -> Response:
+    """Set and save the YODA theme.
+
+    :returns: Redirect to admin page with status message
+    """
+    # Load the selected theme
+    theme = request.form.get('theme')
```

Comment:
Wouldn't hurt to do some input validation here. For example, check that the theme the user gives exists in the map.

Reply:
Added the theme validation